src/clients/inst_disk_proposal.rb

```diff
@@ -0,0 +1,86 @@
+# encoding: utf-8
+
+# Copyright (c) [2016] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may
+# find current contact information at www.novell.com.
+
+require "yast"
+require "y2storage"
+
+Yast.import "UI"
+Yast.import "Wizard"
+Yast.import "HTML"
+
+module Yast
+  #
+  # client to calculate the storage proposal during installation and provide
+  # the user a summary of the storage proposal
+  #
+  class InstDiskProposalClient < Client
+    include Yast::I18n
+    include Yast::Logger
+
+    using Y2Storage::Refinements::Devicegraph
+
+    def main
+      textdomain "storage"
+
+      begin
+        proposal = Y2Storage::Proposal.new(settings: settings)
+        proposal.propose
+        actiongraph = proposal.devices.actiongraph
+        summary = summary(actiongraph)
+      rescue Y2Storage::Proposal::Error
+        log.error("generating proposal failed")
```

Comment:
Logging something is better than logging nothing, but I meant to log the exact exception rescued, otherwise we will be left in the dark, won't we?